bidswitch-json/src/main/scala/com/powerspace/openrtb/bidswitch/bidresponse/BidSwitchBidResponseSerde.scala

```diff
@@ -2,27 +2,39 @@ package com.powerspace.openrtb.bidswitch.bidresponse
 
 import com.google.openrtb.BidResponse
 import com.powerspace.bidswitch.{BidResponseExt, BidswitchProto}
+import com.powerspace.openrtb.bidswitch.util.JsonUtils
+import com.powerspace.openrtb.json.EncoderProvider
+import com.powerspace.openrtb.json.util.EncodingUtils
 import com.powerspace.openrtb.json.bidresponse.OpenRtbBidResponseSerde
+import com.powerspace.openrtb.json.common.OpenRtbProtobufEnumEncoders
 import io.circe.generic.extras.Configuration
 
 /**
   * Decoder for the BidSwitch bid response extension
   */
-object BidSwitchBidResponseSerde {
+object BidSwitchBidResponseSerde extends EncoderProvider[BidResponse] {
 
   import io.circe._
+  import io.circe.syntax._
+  import io.circe.generic.extras.semiauto._
+  import JsonUtils._
+  import EncodingUtils._
   import BidSwitchBidSerde._
-
-  implicit val customConfig: Configuration = Configuration.default.withSnakeCaseMemberNames
+  import OpenRtbProtobufEnumEncoders._
 
   private implicit val bidResponseExtDecoder: Decoder[BidResponseExt] = _.downField("ext")
     .downField("protocol")
     .as[Option[String]]
     .map(BidResponseExt.apply)
+  private implicit val bidResponseExtEncoder: Encoder[BidResponseExt] = deriveEncoder[BidResponseExt].cleanRtb
+
+  private implicit val seatBidEncoder = BidSwitchBidSerde.seatBidEncoder
 
-  implicit def decoder: Decoder[BidResponse] = for {
+  def decoder: Decoder[BidResponse] = for {
     bidResponse <- OpenRtbBidResponseSerde.decoder
```

Comment:
not a good naming (not you but you could give it a better name

Reply:
It think it is the purpose of a *BidrreponseSerde to provide a `Decoder[BidResponse]]` decoder